txclib/api.py

```diff
@@ -0,0 +1,77 @@
+import six
+import requests
+
+from requests.auth import HTTPBasicAuth
+
+from txclib import utils
+from txclib.urls import API_URLS, HOSTNAMES
+from txclib.log import logger
+
+
+class Api(object):
+
+    USERNAME = 'api'
+    VALID_CALLS = ['auth_check', 'projects', 'organizations', 'formats']
+
+    def map_paths_to_hostnames(cls, path_to_tx, host):
+        domains = utils.get_api_domains(path_to_tx, host)
+        return {
+            path: domains[key] for key, paths in HOSTNAMES.items()
+            for path in paths
+        }
+
+    def __init__(self, token=None, username=None, password=None,
+                 path_to_tx=None, host=None):
+        self.hostnames = self.map_paths_to_hostnames(path_to_tx, host)
+        if token:
+            self.token = token
+            self.username = self.USERNAME
+        elif username and password:
+            self.token = password
+            self.username = username
+        else:
+            logger.error("Authorization credentials are missing. Make sure "
+                         "that you have run `tx init` to setup your "
+                         "credentials.")
+
+    def get(self, api_call, *args, **kwargs):
+        """
+        Performs the GET api call specified by api_call and
+        parses the response
+        """
+        # mock response
+        if api_call not in self.VALID_CALLS:
+            raise Exception(
+                "Tried to perform unsupported api call {}".format(
+                    api_call
+                )
+            )
+
+        hostname = self.hostnames[api_call]
+        url = API_URLS[api_call] % kwargs
+        url = "{}{}".format(hostname, url)
+
+        try:
+            response = requests.get(
+                url, auth=HTTPBasicAuth(self.username, self.token)
+            )
+            response.raise_for_status()
+            all_data = utils.parse_json(response.content)
```

Comment:
This fails on python3 because `response.content` is `bytes`, not `str`. Why not use response.json()?

Reply:
Fixed